File: Examples/android/CMeSdkJava/app/src/main/java/me/connect/sdk/java/sample/db/dao/ConnectionDao.java
package me.connect.sdk.java.sample.db.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import me.connect.sdk.java.sample.db.entity.Connection;

@Dao
public interface ConnectionDao {
    @Query("SELECT * FROM connection")
    List<Connection> getAllAsync();

    @Query("SELECT * FROM connection")
    LiveData<List<Connection>> getAll();

    @Query("SELECT serialized FROM connection")
    List<String> getAllSerializedConnections();

    @Query("SELECT * FROM connection WHERE id = :id")
    Connection getById(int id);

    @Query("SELECT * FROM connection WHERE pwdid = :pwDid")
    Connection getByPwDid(String pwDid);

    @Insert
    void insertAll(Connection... connections);

    @Update
    void update(Connection connection);

    @Delete
    void delete(Connection connection);
}

File: examples/android/MSDKSampleAppKotlin/app/src/main/java/msdk/kotlin/sample/db/dao/ConnectionDao.kt
package msdk.kotlin.sample.db.dao


import androidx.lifecycle.LiveData
import androidx.room.*

import msdk.kotlin.sample.db.entity.Connection


@Dao
interface ConnectionDao {
    @Query("SELECT * FROM connection")
    suspend fun getAllAsync(): List<Connection>

    @Query("SELECT * FROM connection")
    fun getAll(): LiveData<List<Connection>>

    @Query("SELECT serialized FROM connection")
    suspend fun getAllSerializedConnections(): List<String>

    @Query("SELECT * FROM connection WHERE id = :id")
    suspend fun getById(id: Int): Connection

    @Query("SELECT * FROM connection WHERE pwdid = :pwDid")
    suspend fun getByPwDid(pwDid: String): Connection

    @Insert
    suspend fun insertAll(vararg connections: Connection)

    @Update
    suspend fun update(connection: Connection)

    @Delete
    suspend fun delete(connection: Connection)
}
